google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/PullSubscriber.java

```diff
@@ -16,11 +16,16 @@
 
 package com.google.cloud.pubsublite.internal;
 
+import com.google.cloud.pubsublite.Offset;
 import io.grpc.StatusException;
 import java.util.List;
+import java.util.Optional;
 
 // A PullSubscriber exposes a "pull" mechanism for retrieving messages.
 public interface PullSubscriber<T> extends AutoCloseable {
   // Pull currently available messages from this subscriber. Does not block.
   List<T> pull() throws StatusException;
+
+  // The next offset expected to be returned by this PullSubscriber, or empty if unknown.
```

Comment:
Should we say something like "subsequent messages are guaranteed to have offsets of at least this value"? Just to make it clear that there can be gaps in Offsets.